### DISCUSSION TITLE
Migration down missing. Intentional?

### DISCUSSION THREAD

**Discussion by TechTailor** [spatie/laravel-medialibrary#2264] [General] [answered]
Okay so I can across this issue when building a new app, needed to reset the migrations (due to changes to several tables) and came to realize that the media table doesn't have the down method. 

Now I know have a down method in a production environment doesn't really make sense and of course, you can always just add the method yourself after publishing the migration but wanted to know if there's a particular reason for not shipping it by default?

**Comment by freekmurze** [comment, marked as answer]
We never migrate down. When we want to revert something we just write a new migration.

**Reply by DanWithams** [reply]
There are arguments for and against using down to manage rollbacks on non-development environments, and I would say "his to his own" in that regard.

However the lack of a down() migration prevents anyone from using the `php artisan migrate:refresh` development cycle. Would you consider a PR in order to support `migrate:refresh` ?

**Reply by TechTailor** [reply]
I ended up publishing the migration and adding the down method myself so as to allow the installer that ships with my app to perform rollback in case of any issues during installation.

**Reply by DanWithams** [reply]
@TechTailor I've done the same for the project I am currently working on. However I'm not looking for a single case solution, I am suggesting an improvement to the maintainer and offering to make the PR for it.
